File: lib/169.majority_element.dart
void main() {
  print(majorityElement([3, 2, 3]));
  print(majorityElement([2, 2, 1, 1, 1, 2, 2]));
}

int majorityElement(List<int> nums) {
  final Map<int, int> map = {};

  nums.forEach((element) {
    final value = map[element] ?? 0;
    map[element] = value + 1;
  });

  var maxKey = 0;
  var maxValue = 0;

  map.forEach((key, value) {
    if (value > maxValue) {
      maxKey = key;
      maxValue = value;
    }
  });

  return maxKey;
}

File: src/169.majority_element.kt
fun main() {
    println(majorityElement(intArrayOf(3, 2, 3)))
    println(majorityElement(intArrayOf(2, 2, 1, 1, 1, 2, 2)))
}

fun majorityElement(nums: IntArray): Int {
    val map = mutableMapOf<Int, Int>()

    nums.forEach { e ->
        map[e] = (map[e] ?: 0) + 1
    }

    var maxKey = -1
    var maxValue = -1

    map.forEach { (k, v) ->
        if (v > maxValue) {
            maxKey = k
            maxValue = v
        }
    }

    return maxKey
}